src/core/src/op/scatter_elements_update.cpp

```diff
@@ -78,256 +71,127 @@ shared_ptr<Node> op::v12::ScatterElementsUpdate::clone_with_new_inputs(const Out
-
-    return scatter_elements_update::evaluate_scatter_elements_update(inputs[0],
-                                                                     inputs[1],
-                                                                     inputs[2],
-                                                                     inputs[3],
-                                                                     outputs[0],
-                                                                     normalized_axis);
-}
-
-bool op::v12::ScatterElementsUpdate::evaluate_scatter_elements_update(const HostTensorVector& outputs,
-                                                                      const HostTensorVector& inputs) const {
-    const auto normalized_axis = get_normalized_axis(inputs);
-
-    return scatter_elements_update::evaluate_scatter_elements_update(inputs[0],
-                                                                     inputs[1],
-                                                                     inputs[2],
-                                                                     inputs[3],
-                                                                     outputs[0],
-                                                                     normalized_axis,
-                                                                     m_reduction,
-                                                                     m_use_init_val);
-}
-
-bool op::v3::ScatterElementsUpdate::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool op::v3::ScatterElementsUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v3_ScatterElementsUpdate_evaluate);
-    return evaluate_scatter_elements_update(outputs, inputs);
-}
-
-bool op::v12::ScatterElementsUpdate::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+    OPENVINO_ASSERT(inputs.size() == 4);
+    OPENVINO_ASSERT(outputs.size() == 1);
+
+    const auto& data = inputs[0];
+    const auto& indices = inputs[1];
+    const auto& updates = inputs[2];
+    auto& output = outputs[0];
+    const auto& data_shape = data.get_shape();
+    const auto& indices_shape = indices.get_shape();
+    output.set_shape(data_shape);
+    using namespace ov::element;
+    return IfTypeOf<boolean, f16, f32, i16, i32, i64, u32, u64>::apply<scatter_elements_update::Evaluate>(
+        data.get_element_type(),
+        data,
+        indices,
+        updates,
+        output,
+        data_shape,
+        indices_shape,
+        get_normalized_axis(inputs),
+        op::v12::ScatterElementsUpdate::Reduction::NONE,
+        false);
+}
+
+bool op::v12::ScatterElementsUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v12_ScatterElementsUpdate_evaluate);
-    return evaluate_scatter_elements_update(outputs, inputs);
+    OPENVINO_ASSERT(inputs.size() == 4);
```

Comment:
Use shape_infer to do validation, is more complex then just this check and set output shape as input but,  `shape_infer` check others shapes and axis value then data provided to evaluate should correct sizes.

New version of `get_normalized_axis` dont validate axis value.

Reply:
As agreed offline, "stronger" version of axis normalization is enough to cover safety issues.